docs/v1/open_questions.md

```diff
@@ -0,0 +1,112 @@
+# Open Questions
+
+This document contains open questions regarding the
+[API design document](./design.md).
+
+## 1. Is the `ServiceInstanceBinding` resource necesssary?
+
+The `ServiceInstanceBinding` object is a byproduct of the successful execution
+of a `ServiceInstanceClaim`. Instead of writing a new `ServiceInstanceBinding`
+object, can the service catalog controller just add the binding information
+directly into the `ServiceInstanceClaim` on success?
+
+## Should we write Kubernetes references for binding information?
+
+Currently, the information in a `ServiceInstanceBinding` that points to created
+resources looks like this:
+
+```yaml
+secrets:
+  - my-zip-code-creds-password
+configMaps:
+  - my-zip-code-creds-conn-info
+```
+
+Instead, should we write pointers to these resources using Kubernetes reference
+notation, as below?
+
+```yaml
+resources:
+  - /api/v1/namespaces/${NAMESPACE}/secrets/my-zip-code-creds-password
+  - /api/v1/namespaces/${NAMESPACE}/configmaps/my-zip-code-creds-conn-info
+```
+
+## 2. Should `ServiceInstance`s be namespaced?
```

Comment:
I imagine the vast majority of service instances in a multi-tenant cluster would be unshared (private) or privately shared (among a few namespaces; semi-private). Only a few would be broadly shared, with the desire to be easily discoverable.

Imagine a cluster with 20000 users. Presumably many of these users will want to create one or more private or semi-private service instances. 

If ServiceInstance were not namespaced, then users would not be able to choose the names of their service instances. As with PersistentVolumes, the service instances would have to be automatically created. That would make it impossible to manually create declarative claims/bindings against them, which would be a problem for shared service instances.

A handful of cluster-wide service instances with hand-picked names could be created by a cluster administrator, or by the service administrators, since large shared services are likely to have their own administrators, distinct from the cluster administrator.

So primarily it's the semi-private instances that are a problem with the non-namespaced approach.

OTOH, the main disadvantage of namespaced service instances is discoverability of cluster-wide service instances. However, that could be addressed by choosing a well known namespace by convention, similar to how we use `kube-system`. 

Therefore, always namespacing ServiceInstance seems cleaner to me.

FWIW, that's also roughly analogous to what we do in Borg.